drivers/ethernet/eth_enc28j60.c

```diff
@@ -757,6 +763,23 @@ static enum ethernet_hw_caps eth_enc28j60_get_capabilities(const struct device *
 		;
 }
 
+static int eth_enc28j60_set_config(const struct device *dev,
+	enum ethernet_config_type type,
+	const struct ethernet_config *config)
+{
+	struct eth_enc28j60_runtime *context = dev->data;
+
+	if (type == ETHERNET_CONFIG_TYPE_MAC_ADDRESS) {
+		memcpy(context->mac_address, config->mac_address.addr,
+			MIN(sizeof(context->mac_address), sizeof(config->mac_address.addr)));
```

Comment:
```suggestion
			sizeof(config->mac_address.addr));
```
the min is not really needed

Reply:
Yeah, I guess not ... it is probably optimised out anyway. I was going for full defense. In this case I know the sizes are the same, but since the types aren't ... but the upstream may change, which would probably break in other ways, but we won't have a hiding buffer overrun vulnerability. Added a build assert on the buffer sizes, to keep myself happy :)